tests/integration/cairo1/parser_test.py

```diff
@@ -16,23 +16,38 @@ def test_suite_json_fixture(datadir: Path) -> str:
 
 
 def test_parse(mocker: MockerFixture, test_suite_json: str):
-    test_suite = parse_test_suite(Path("test_source.cairo"), test_suite_json)
-    cheat_mock = mocker.MagicMock()
-    cheat_mock.return_value = 0
-    for case in test_suite.test_cases:
-        runner = CairoFunctionRunner(program=test_suite.program, layout="all")
-        runner.run_from_entrypoint(
-            case.offset,
-            *[],
-            hint_locals={"roll": cheat_mock},
-            static_locals={
-                "__find_element_max_size": 2**20,
-                "__squash_dict_max_size": 2**20,
-                "__keccak_max_size": 2**20,
-                "__usort_max_size": 2**20,
-                "__chained_ec_op_max_len": 1000,
-            },
-            run_resources=RunResources(n_steps=100000000000000000),
-            verify_secure=False,
-        )
-    assert cheat_mock.call_count == 6
+    for mocked_error_code in [0, 1]:
+        test_suite = parse_test_suite(Path("test_source.cairo"), test_suite_json)
+
+        cheat_mock = mocker.MagicMock()
+        cheat_mock.return_value = type(
+            "return_value", (object,), {"err_code": mocked_error_code}
+        )()
+
+        declare_mock = mocker.MagicMock()
+        ok = type("ok", (object,), {"class_hash": 0})()
+        declare_mock.return_value = type(
+            "return_value", (object,), {"err_code": mocked_error_code, "ok": ok}
+        )()
+
+        for case in test_suite.test_cases:
+            runner = CairoFunctionRunner(program=test_suite.program, layout="all")
+            runner.run_from_entrypoint(
+                case.offset,
+                *[],
+                hint_locals={
+                    "roll": cheat_mock,
+                    "declare": declare_mock,
+                    "start_prank": cheat_mock,
+                },
+                static_locals={
+                    "__find_element_max_size": 2**20,
+                    "__squash_dict_max_size": 2**20,
+                    "__keccak_max_size": 2**20,
+                    "__usort_max_size": 2**20,
+                    "__chained_ec_op_max_len": 1000,
+                },
+                run_resources=RunResources(n_steps=100000000000000000),
+                verify_secure=False,
+            )
+            assert mocked_error_code == runner.get_return_values(3)[0]
```

Comment:
I couldn't find a less hacky way of checking the result for now :/

Reply:
We could probably do it like it is done here:
https://github.com/software-mansion-labs/cairo/blob/aba4b69ef925214c77ed57672313ca25f9de4014/crates/cairo-lang-runner/src/lib.rs#L106

Reply:
From what I understand this is basically the same as [here](https://github.com/starkware-libs/cairo-lang/blob/de741b92657f245a50caab99cfaef093152fd8be/src/starkware/cairo/common/cairo_function_runner.py#L185). I guess the function "signature" (rust) may be equivalent to the "identifiers" (py) which we're missing, so we are unable to automatically retrieve the return data, but it's still there in the memory.

Reply:
I'm going to try to solve this

Reply:
I don't think it's quickly doable but might've missed something. I think we should unblock for now https://github.com/software-mansion/protostar/issues/1519

Reply:
Sure